internal/idle/idle.go

```diff
@@ -203,22 +176,46 @@ func (m *manager) tryEnterIdleMode() bool {
 	// No new RPCs have come in since we last set the active calls count value
 	// -math.MaxInt32 in the timer callback. And since we have the lock, it is
 	// safe to enter idle mode now.
-	if err := m.enforcer.EnterIdleMode(); err != nil {
-		m.logger.Errorf("Failed to enter idle mode: %v", err)
-		return false
-	}
+	m.enforcer.EnterIdleMode()
 
 	// Successfully entered idle mode.
 	m.actuallyIdle = true
 	return true
 }
 
+func (m *Manager) EnterIdleModeForTesting() {
```

Comment:
The only reason for having this function defined in the internal package was to be able to test this functionality from a test (wherein, the actual functionality is not exported). Now, with this change, we have a different code path for testing and a different one for non-testing workloads. This makes me feel nervous because our tests are now using a different code and also, we need to ensure that we maintain this code to be in sync with the code that gets run to enter idle mode in real workloads.